### PR TITLE
[Windows] Fix TimeZone display name and daylight saving time check

### PR DESCRIPTION
**Pull request by lxbndr**
This fixes TestDateFormatter.test_expectedTimeZone either.

### CODE REVIEW

**Comment by lxbndr** [comment]
cc @compnerd

File: CoreFoundation/NumberDate.subproj/CFTimeZone.c [hunk 1]
```diff
@@ -1678,10 +1679,22 @@ CFStringRef CFTimeZoneCopyAbbreviation(CFTimeZoneRef tz, CFAbsoluteTime at) {
 }
 
 Boolean CFTimeZoneIsDaylightSavingTime(CFTimeZoneRef tz, CFAbsoluteTime at) {
-    CFIndex idx;
     __CFGenericValidateType(tz, CFTimeZoneGetTypeID());
+#if TARGET_OS_WIN32
+    UErrorCode status = U_ZERO_ERROR;
+    UCalendar *ucal = __CFCalendarCreateUCalendar(NULL, CFSTR("C"), tz);
+    if (ucal == NULL) {
+      return NULL;
```
Comment (spevans) [inline CoreFoundation/NumberDate.subproj/CFTimeZone.c:1687]: Should this not be returning `false` as the return type is `Boolean` ?

Comment (lxbndr) [reply CoreFoundation/NumberDate.subproj/CFTimeZone.c:1687]: Oh, right.

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by millenomi** [comment]
Waiting for @compnerd to yes.

File: CoreFoundation/NumberDate.subproj/CFTimeZone.c [hunk 2]
```diff
@@ -1678,10 +1679,22 @@ CFStringRef CFTimeZoneCopyAbbreviation(CFTimeZoneRef tz, CFAbsoluteTime at) {
 }
 
 Boolean CFTimeZoneIsDaylightSavingTime(CFTimeZoneRef tz, CFAbsoluteTime at) {
-    CFIndex idx;
     __CFGenericValidateType(tz, CFTimeZoneGetTypeID());
+#if TARGET_OS_WIN32
+    UErrorCode status = U_ZERO_ERROR;
+    UCalendar *ucal = __CFCalendarCreateUCalendar(NULL, CFSTR("C"), tz);
+    if (ucal == NULL) {
+      return false;
```
Comment (compnerd) [inline CoreFoundation/NumberDate.subproj/CFTimeZone.c:1687]: I think that we should spell this as FALSE as it is the CF Boolean type, not guaranteed to be _Bool

File: CoreFoundation/NumberDate.subproj/CFTimeZone.c [hunk 3]
```diff
@@ -1678,10 +1679,22 @@ CFStringRef CFTimeZoneCopyAbbreviation(CFTimeZoneRef tz, CFAbsoluteTime at) {
...
+    }
+    ucal_setMillis(ucal, (at + kCFAbsoluteTimeIntervalSince1970) * 1000.0, &status);
+
+    UBool isDaylightTime = ucal_inDaylightTime(ucal, &status);
+    return isDaylightTime;
```
Comment (compnerd) [inline CoreFoundation/NumberDate.subproj/CFTimeZone.c:1692]: This should be converted from the ICU Boolean to the CoreFoundation Boolean type

**Comment by spevans** [comment]
@swift-ci test linux

File: CoreFoundation/NumberDate.subproj/CFTimeZone.c (hunk 3 above)
Comment (lxbndr) [reply CoreFoundation/NumberDate.subproj/CFTimeZone.c]: Not sure `return isDaylightTime ? TRUE : FALSE` is the preferred way. But should be safest though.

File: CoreFoundation/NumberDate.subproj/CFTimeZone.c (hunk 2 above)
Comment (lxbndr) [reply CoreFoundation/NumberDate.subproj/CFTimeZone.c]: Done

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
Windows Run: https://dev.azure.com/compnerd/swift-build/_build/results?buildId=29766&view=results

**Comment by compnerd** [comment]
Windows looks good with this :)

**Pull request merged**